operator/pkg/utils/pki/pki.go

```diff
@@ -0,0 +1,196 @@
+/*
```

Comment:
I think we'd get a lot of mileage from creating abstraction (structs) that encapsulate these concepts. Something along the lines of:
```
type KeyPair {
  Certificate ...
  PrivateKey ...
}

type CertificateAuthority {
   Name
   KeyPair
}

type Certificate struct {
  common name
  ...
}
```
Of course, if libraries we're relying on already have good concepts for them, let's re-use them, by all means. I want to avoid cases where we're returning multiple return types of primitives, as it requires deep understanding of the code to understand what's going on. 

Feel free to punt on all of this -- we can refactor later if you'd prefer.

Reply:
I will revisit this, in favor of time and bias for action, I will address this later once we have the functionality.